### PR TITLE
Adding support for scenario based regional emission targets

### PR DESCRIPTION
**Pull request by Renato-Rodrigues**
## Purpose of this PR

- Add support for scenario based regional emission targets. 
  - This allows to define in the GAMS code instead of the scenario config files more complex scenarios that include multiple regional emission targets. 
  - It avoids the 255 character line limitation of GAMS for special scenario cases.
  - In order to define a hard-coded scenario option, please include the scenario definition to the module `47_regipol\regiCarbonPrice\declarations.gms` file.
  - In order to use a hard-coded scenario, set the `cm_emiMktTarget` value to the scenario name. E.g. `cm_emiMktTarget` = `nzero`.

- Add scenario to reflect declared net-zero targets from countries explicitly handled by the model.
  - This scenario is named `nzero` and it is used for the IAMC National Scenarios call.
  - Source of net-zero target pledges: https://zerotracker.net/
  - Targets added to the `nzero` scenario:

Region | 2030 target | net-zero year | gasses_coverage
-- | -- | -- | --
EU27_regi | 57% (1990) | 2050 | GHG
DEU | 65% (1990) | 2045 | GHG
UKI |   | 2050 | GHG
USA |   | 2050 | GHG
JPN |   | 2050 | GHG
CHA |   | 2060 | CO2
IND |   | 2070 | CO2

- Possible improvement to be considered in the `nzero` scenario definition
  i. Consider 2030 targets for all the regions above. 
  - This requires calculating baseline emissions for defining the targets in the different regions.
 
Region | 2030 target
-- | --
UKI | 68% (1990)
USA | 50-52% (2005)
JPN | 46% (2013)
CHA | 65% (2005, per unit of GDP)
IND | 45% (2005)

   ii. Consider aggregated regions targets: see @orichters comment below about CAZ.   

## Type of change

- [x] New feature 

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [x] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

## Further information (optional):

* Test runs are here: 

-  `/p/projects/ecemf/REMIND/2040_scenarios/v06_2024_05_14_rev1_SWG/output/06_iamcSWG_2024-05-18_16.15.12`

### CODE REVIEW

**Comment by Renato-Rodrigues** [comment]
@orichters and @cchrisgong As I think you two work with similar scenario definitions, would you have any comment, critique or suggestions about the above described `nzero` regional emissions target scenario?

**Comment by Renato-Rodrigues** [comment]
@robertpietzcker The run using the above-mentioned `nzero` scenario definition is still in progress, but it seems that the challenge of achieving convergence lies not in the regional net zero targets (only India is experiencing a slight issue in reaching the tolerance I set, but this could probably be resolved), but rather in achieving convergence in primary energy trade.

Since this is an "extreme" scenario, I expected a worst-case situation. However, as debugging the primary energy trade is more challenging, I just wanted to ping you on this in case you have any immediate thoughts on this.

I will create a `nashAnalysis` report as soon as the run finishes to further analyze the situation.

File: main.gms [hunk 1]
```diff
@@ -1260,6 +1260,9 @@ $setGlobal cm_regiExoPrice  off    !! def = off
 ***   Example on how to use:
 ***     cm_emiMktTarget = '2020.2050.EU27_regi.all.budget.netGHG_noBunkers 72, 2020.2050.DEU.all.year.netGHG_noBunkers 0.1'
 ***     sets a 72 GtCO2eq budget target for European 27 countries (EU27_regi), for all GHG emissions excluding bunkers between 2020 and 2050; and a 100 MtCO2 CO2eq emission target for the year 2050, for Germany"
+***     cm_emiMktTarget = 'nzero'
+***     loads hard-coded options for regional target scenarios defined in the module '47_regipol\regiCarbonPirce' declarations file.
```
Comment (orichters) [inline main.gms:1264]:
```suggestion
***     loads hard-coded options for regional target scenarios defined in the module '47_regipol/regiCarbonPrice' declarations file. 
```

**Review by orichters — COMMENTED** [review]
Well, we basically implement net Zero targets additionally (via a tax markup) to global carbon prices (such as a 1.5K or 2K scenario using `diffCurvPhaseIn2Lin`). Also, my assessment was that China has a GHG, not CO2target.

**Comment by Renato-Rodrigues** [comment]
Thk you Oliver. 

> Well, we basically implement net Zero targets additionally (via a tax markup) to global carbon prices (such as a 1.5K or 2K scenario using `diffCurvPhaseIn2Lin`). 

Yes, I am aware. This is an alternative way of doing the targets.

> Well, we basically implement net Zero targets additionally (via a tax markup) to global carbon prices (such as a 1.5K or 2K scenario using `diffCurvPhaseIn2Lin`). Also, my assessment was that China has a GHG, not CO2target.

I am following in principle the classification made by the net zero tracker team. For China they state:

```
Coverage of gases: The phraseology in the Chinese NDC and Mid-Century Long-Term Low Greenhouse Gas Emission Development Strategy (LTS) refers only to "carbon" neutrality (碳中和). 
Although Chinese Special Climate Envoy Xie Zhenhua stated, in mid-2021, that the 2060 neutrality target will cover all greenhouse gas emissions (not just CO2) the LTS has explicit reference 'carbon neutrality' and links to the 2030 emissions peaking target which relates only to carbon emissions.  This coding preferences the official documents over the government announcement. 

Special Climate Envoy Xie Zhenhua references (https://chinadialogue.net/en/digest/2060-neutrality-pledge-includes-all-greenhouses-gases/) or in Chinese (http://www.ncsc.org.cn/xwdt/gnxw/202107/t20210727_851433.shtml). 
```

I am open to adjust this if we want an integrated definition, but as I do not speak Chinese, I tend to follow their assessment instead. Maybe @cchrisgong could provide her point of view about that?

**Comment by cchrisgong** [comment]
hey, I see the discussion is too long, so I moved it to gitlab

**Review by orichters — COMMENTED** [review]
Note that Canada, Australia and Net Zealand all have 2050 net Zero GHG targets, as well as the majority of LAM countries. In NGFS, we also add LAM and CAZ to the 2050 list.

**Comment by Renato-Rodrigues** [comment]
> Note that Canada, Australia and Net Zealand all have 2050 net Zero GHG targets, as well as the majority of LAM countries. In NGFS, we also add LAM and CAZ to the 2050 list.

Thank you @orichters 

As soon as I reach an stable solution with the net-zero targets in countries explicitly handled by the model, I could extend the formulation to include aggregated regions as you mentioned above.

For now I will keep this scenario example as is, as it is already enough challenging to make sure that the model can handle well this case with the scenario configs that I am currently trying.

**Comment by cchrisgong** [comment]
cc @lecfab

**Pull request merged**